app/services/participants/change_schedule/ecf.rb

```diff
@@ -3,17 +3,31 @@
 module Participants
   module ChangeSchedule
     class ECF
-      include ProfileAttributes
       include ActiveModel::Validations
 
+      attr_accessor :course_identifier, :participant_id, :cpd_lead_provider
+
       delegate :school_cohort, to: :user_profile, allow_nil: true
 
+      validates :course_identifier, presence: { message: I18n.t(:missing_course_identifier) }
```

Comment:
Would it make sense to have an `inclusion` validation rule here?

Reply:
> Looks good, left one minor comment on the validation of course_identifier (vs doing it in #valid_courses), but there's a chance I'm just missing the why.

what you said is perfectly reasonable and this class still needs more refining. i feel now at least things start to become more sane so things like this are more visible and changes like your suggestions can be made as the code becomes easier to reason with